tests/tcam_mmio.h

```diff
@@ -0,0 +1,46 @@
+#ifndef __TCAM_MMIO_H__
+#define __TCAM_MMIO_H__
+
+#include <stdint.h>
+#include <stdio.h>
+#include "mmio.h"
+
+// TCAM register base and offsets
+#define TCAM_BASE      0x4000
+#define TCAM_STATUS    (TCAM_BASE + 0x00)
+#define TCAM_CONTROL   (TCAM_BASE + 0x04)
+#define TCAM_WDATA     (TCAM_BASE + 0x08)
+#define TCAM_ADDRESS   (TCAM_BASE + 0x0C)
+
+// Delay functions for timing
+void delay_write() {
+    for (volatile int i = 0; i < 1; i++); 
+}
+
+void delay_read() {
+    for (volatile int i = 0; i < 2; i++); 
+}
+
+// Write a value to the TCAM at a given address
+// csb=0, web=0, wmask=0xF
+void write_tcam(uint32_t data, uint32_t address) {
```

Comment:
this method is good one, but there should be another method, such as write_tcam_list or something, to write the entire array of data directly using a singular function.